app/registration/api.py

```diff
@@ -44,7 +44,50 @@ def api_register():
         data = registration.blob,
     )
 
-@registration_blueprint.route('/api/order/token')
+@registration_blueprint.route('/api/priorities', methods=['GET', 'POST'])
+@oauth.require_oauth('registration_priorities')
 @token_auth.login_required
-def validate_token():
-    return jsonify(g.uni.name)
+def api_registration_priorities():
+    if request.method == 'POST' \
+        and request.headers.get('Content-Type') == 'application/json':
+        req = request.get_json()
+        if not req:
+            abort(403)
+            return ''
+
+        new_priority = len(req['priority'])
+
+        registrations = Registration.query.filter_by(uni_id = g.uni.id)
+        for registration in registrations:
```

Comment:
TBH, I still don't see how this cannot just be:
```python
# Get a list of all registrations by this uni
registrations = {reg.id: reg for reg in Registration.query.filter_by(uni_id = g.uni.id)}

# Set the priorities according to the order they are given in in the confirmed list
for priority, reg_id in enumerate(req['confirmed']):
    reg = registrations.pop(reg_id, None)
    # FIXME: error handling for the case reg == none
    reg.confirmed = True
    reg.priority = priority
    db.session.add(reg)

# Set all remaining priorities as unconfirmed
for reg in registrations:
    reg.confirmed = False
    reg.priority = None
    db.session.add(reg)
```

Additionally, we might want to add a check for negative priorities (guaranteed places).